File: backend/models/file.py
from bson import ObjectId
from datetime import datetime, timezone
from typing import Optional

def create_file_document(*, owner_id: ObjectId, cluster_id: Optional[ObjectId], original_name: str, mime: str, extension: str, bucket: str, key: str, size_bytes: int, region: Optional[str] = None, visibility: str = "private",
    checksum: Optional[str] = None,) -> dict:

    return {

        "_id": ObjectId(),

        "ownerId": owner_id,
        "clusterId": cluster_id,

        "originalName": original_name,
        "mime": mime,
        "extension": extension,

        "bucket": bucket,
        "key": key,
        "region": region,
        "sizeBytes": size_bytes,
        "checksum": checksum,

        "visibility": visibility,
        "status": "uploading",

        "viewCount": 0,
        "downloadCount": 0,

        "isDeleted": False,
        "deletedAt": None,

        "createdAt": datetime.now(timezone.utc),
        
    }

File: backend/src/main/java/com/filespark/server/models/File.java
package com.filespark.server.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "files")
public class File {

    @Id
    private String id;

    private String ownerId;
    private String clusterId;

    private String originalName;
    private String mime;
    private String extension;

    private String bucket;
    private String key;
    private String region;
    private long sizeBytes;
    private String checksum;

    private String visibility;
    private String status;

    private long viewCount;
    private long downloadCount;

    private boolean isDeleted;
    private Instant deletedAt;

    private Instant createdAt;

    public File(String ownerId, String clusterId, String originalName, String mime, String extension, String bucket, String key, long sizeBytes, String region,  String visibility, String checksum) {

        this.ownerId = ownerId;
        this.clusterId = clusterId;

        this.originalName = originalName;
        this.mime = mime;
        this.extension = extension;

        this.bucket = bucket;
        this.key = key;
        this.region = region;
        this.sizeBytes = sizeBytes;
        this.checksum = checksum;

        this.visibility = visibility != null ? visibility : "private";
        this.status = "uploading";

        this.viewCount = 0;
        this.downloadCount = 0;

        this.isDeleted = false;
        this.deletedAt = null;

        this.createdAt = Instant.now();

    }

    public String getId() {

        return id;
        
    }

    public String getOwnerId() {

        return ownerId;

    }

    public String getClusterId() {

        return clusterId;

    }

    public String getOriginalName() {

        return originalName;

    }

    public String getMime() {

        return mime;

    }

    public String getExtension() {

        return extension;

    }

    public String getBucket() {

        return bucket;

    }

    public String getKey() {

        return key;

    }

    public String getRegion() {

        return region;

    }

    public long getSizeBytes() {

        return sizeBytes;

    }

    public String getChecksum() {

        return checksum;

    }

    public String getVisibility() {

        return visibility;

    }

    public String getStatus() {

        return status;

    }

    public long getViewCount() {

        return viewCount;

    }

    public long getDownloadCount() {

        return downloadCount;

    }

    public boolean isDeleted() {

        return isDeleted;

    }

    public Instant getDeletedAt() {

        return deletedAt;

    }

    public Instant getCreatedAt() {

        return createdAt;

    }

}
